src/Mjolnir.ts

```diff
@@ -1023,9 +1023,9 @@ export class Mjolnir {
         return await this.eventRedactionQueue.process(this, roomId);
     }
 
-    private async handleReport(roomId: string, reporterId: string, event: any, reason?: string) {
-        for (const protection of this.enabledProtections) {
-            await protection.handleReport(this, roomId, reporterId, event, reason);
+    private async handleReport(e: { mjolnir: Mjolnir, roomId: string, reporterId: string, event: any, reason?: string }) {
```

Comment:
I don't think we need `mjolnir: Mjolnir` anymore, do we?

Now, `this` should work nicely.

Reply:
Apparently using `this` and not passing the Mjolnir object through the event still has the traceback happening. Just tested.